Lib/fontTools/ttLib/tables/TupleVariation.py

```diff
@@ -614,67 +614,50 @@ def compileTupleVariationStore(variations, pointCount,
 	if len(variations) == 0:
 		return (0, b"", b"")
 
-	# Each glyph variation tuples modifies a set of control points. To
-	# indicate which exact points are getting modified, a single tuple
-	# can either refer to a shared set of points, or the tuple can
-	# supply its private point numbers.  Because the impact of sharing
-	# can be positive (no need for a private point list) or negative
-	# (need to supply 0,0 deltas for unused points), it is not obvious
-	# how to determine which tuples should take their points from the
-	# shared pool versus have their own. Perhaps we should resort to
-	# brute force, and try all combinations? However, if a glyph has n
-	# variation tuples, we would need to try 2^n combinations (because
-	# each tuple may or may not be part of the shared set). How many
-	# variations tuples do glyphs have?
-	#
-	#   Skia.ttf: {3: 1, 5: 11, 6: 41, 7: 62, 8: 387, 13: 1, 14: 3}
-	#   JamRegular.ttf: {3: 13, 4: 122, 5: 1, 7: 4, 8: 1, 9: 1, 10: 1}
-	#   BuffaloGalRegular.ttf: {1: 16, 2: 13, 4: 2, 5: 4, 6: 19, 7: 1, 8: 3, 9: 8}
-	#   (Reading example: In Skia.ttf, 41 glyphs have 6 variation tuples).
-	#
-
-	# Is this even worth optimizing? If we never use a shared point
-	# list, the private lists will consume 112K for Skia, 5K for
-	# BuffaloGalRegular, and 15K for JamRegular. If we always use a
-	# shared point list, the shared lists will consume 16K for Skia,
-	# 3K for BuffaloGalRegular, and 10K for JamRegular. However, in
-	# the latter case the delta arrays will become larger, but I
-	# haven't yet measured by how much. From gut feeling (which may be
-	# wrong), the optimum is to share some but not all points;
-	# however, then we would need to try all combinations.
-	#
-	# For the time being, we try two variants and then pick the better one:
-	# (a) each tuple supplies its own private set of points;
-	# (b) all tuples refer to a shared set of points, which consists of
-	#     "every control point in the glyph that has explicit deltas".
```

Comment:
I see you removed this long block of comments, as I guess no longer applies to the code you modified. I don't fully grasp the consequences of your change. Is it going to change the way TupleVariations gets compiled to binary? Could it affect the size of the table? I should probably refresh my memory by rereading the spec. Would be nice if you added some words (can be shorter that the verbose comment above) about how this is different from before.

Reply:
I'll add a comment.

> Is it going to change the way TupleVariations gets compiled to binary? Could it affect the size of the table?

Fore pretty much all real cases it doesn't make a difference. For very esoteric designspace situations the new code can use the sharedPoints whereas the old code couldn't.

Reply:
It might make iup-optimized fonts smaller though. Let me test with NotoSans.

Reply:
Honestly I'm struggling to find what to comment there. The old code was adhoc and wrong. The new code is exact: it collects all pointSets used and picks the one to share that saves most bytes...

As a headsup, to truly do things properly we should move the IUP optimizer into this compile code. I'll head in that direction...

Reply:
It does:
```
1264516 bytes 15 seconds this branch
1270936 bytes 21 seconds main branch
```

Reply:
Added a comment.